### DISCUSSION TITLE
How many locals is ok? R0914 vs. WPS210

### DISCUSSION THREAD

**Discussion by cidrblock** [ansible/ansible-navigator#702] [Linters, linting, formatting, pre-commit, CI]
WPS210 defaults to 5:

```
lint-vetting: WPS210 Found too many local variables: 6 > 5
```
(https://wemake-python-stylegui.de/en/latest/pages/usage/violations/complexity.html#wemake_python_styleguide.violations.complexity.TooManyLocalsViolation)

but pylint is set to 16:

```
R0914: Too many local variables (16/15) (too-many-locals)
```
(http://pylint-messages.wikidot.com/messages:r0914)

I believe they are checking the same thig, but have different thresholds

**Comment by webknjaz** [comment]
In general, a smaller amount of locals results in easier-to-understand smaller functions that may also be composable and most likely pure. So lower is better. But depending on the project it may make sense to use different defaults sometimes if it's hard to produce simple code. At least, temporarily, that is.

Can you link the reasoning both linters have for this?

**Reply by ssbarnea** [reply]
I propose to disable anything that wps has that overlaps or conflicts with pylint features.

Main reason for this is that pylint is far more popular, older and better known.

I hope that that list is not very long.
